Visualizer/plotTensor.m

```diff
@@ -26,75 +26,105 @@ function plotTensor(tensor, alpha, slicedPlanes, sliceLocations)
+    error('sliceLocations are outside the tensor dimensions.')
 end
 
-% Check tensor type
-if strcmpi(tensor.type, "Metric")
-    titleCharacter = "g";
-elseif strcmpi(tensor.type, "Stress-Energy")
-    titleCharacter = "T";
-end
+% Determine tensor type and index for title
+titleCharacter = switch lower(tensor.type)
+    case "metric"
+        "g"
+    case "stress-energy"
+        "T"
+    otherwise
+        error("Unknown tensor type. Must be 'Metric' or 'Stress-Energy'.")
+end;
 
-% Check tensor index
-if strcmpi(tensor.index, "covariant")
-    titleAugment1 = "_{";
-    titleAugment2 = "";
-elseif strcmpi(tensor.index, "contravariant")
-    titleAugment1 = "^{";
-    titleAugment2 = "";
-elseif strcmpi(tensor.index, "mixedupdown")
-    titleAugment1 = "^{";
-    titleAugment2 = "}_{ ";
-elseif strcmpi(tensor.index, "mixeddownup")
-    titleAugment1 = "_{";
-    titleAugment2 = "}^{ ";
+titleAugment1 = "";
+titleAugment2 = "";
+switch lower(tensor.index)
+    case "covariant"
+        titleAugment1 = "_{";
+        titleAugment2 = "}";
+    case "contravariant"
+        titleAugment1 = "^{";
+        titleAugment2 = "}";
+    case "mixedupdown"
+        titleAugment1 = "^{";
+        titleAugment2 = "}_{";
+    case "mixeddownup"
+        titleAugment1 = "_{";
+        titleAugment2 = "}^{";
+    otherwise
+        error("Unknown tensor index. Must be 'covariant', 'contravariant', 'mixedupdown', or 'mixeddownup'.")
 end
 
 % Check that the coords are cartesian
 if strcmpi(tensor.coords, "cartesian")
     [xLabelText, yLabelText] = labelCartesianAxis(slicedPlanes);
-
-    if strcmpi(tensor.index, "mixedupdown") || strcmpi(tensor.index, "mixeddownup")
-        c1 = [1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4];
-        c2 = [1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4];
-    else
-        c1 = [1, 1, 1, 1, 2, 3, 4, 2, 2, 3];
-        c2 = [1, 2, 3, 4, 2, 3, 4, 3, 4, 4];
-    end
-
+    
+    % Determine coordinates to plot
+    c1 = [1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4];
+    c2 = [1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4];
+    
+    % Get slice data
     idx = getSliceData(slicedPlanes, sliceLocations, tensor);
-
+    
+    % Plot each component
     for i = 1:length(c1)
-        plotComponent(squeeze(tensor.tensor{c1(i), c2(i)}(idx{1}, idx{2}, idx{3}, idx{4}))', titleCharacter+titleAugment1+num2str(c1(i))+titleAugment2+num2str(c2(i))+"}", xLabelText, yLabelText, alpha)
+        component = squeeze(tensor.tensor{c1(i), c2(i)}(idx{1}, idx{2}, idx{3}, idx{4}));
+        plotComponent(component', ...
+            [titleCharacter, titleAugment1, num2str(c1(i)), titleAugment2, num2str(c2(i)), "}"], ...
+            xLabelText, yLabelText, alpha);
     end
 else
-    error('Unknown coordinate system, must be: "cartesian"')
+    error('Unknown coordinate system. Must be: "cartesian"')
+end
+
+end
+
+% Helper function to label Cartesian axes
+function [xLabelText, yLabelText] = labelCartesianAxis(slicedPlanes)
```

Comment:
These are placeholders. Make sure to review the AI-generated code actually provides value.